chef/data_bags/crowbar/migrate/dns/003_add_element_run_list_order.rb

```diff
@@ -1,9 +1,20 @@
 def upgrade ta, td, a, d
   d['element_run_list_order'] = td['element_run_list_order']
+
+  # Make sure that all dns-servers will refresh roles order for run_list
+  nodes = NodeObject.find('roles:dns-server')
+  nodes.each do |node|
+    node.delete_from_run_list('dns-client')
+    node.add_to_run_list('dns-client',
+                         td['element_run_list_order']['dns-client'],
+                         td['element_states']['dns-client'])
+    node.save
+  end
```

Comment:
There are two bugs here:

 - if a dns-server node doesn't have the dns-client role (okay, unlikely)
 - we need to do that for all dns-client nodes as they might get a dns-server role later on